File: migrate_urls.js

const fs = require('fs').promises;
const path = require('path');

const cwd = 'c:/Users/Thinkpad/Downloads/Heal_360';
const filesToMigrate = [
    'about.html',
    'program.html',
    'team.html',
    'blog.html',
    'contact.html',
    'privacy.html',
    'blog-details.html'
];

async function migrate() {
    for (const file of filesToMigrate) {
        const filePath = path.join(cwd, file);
        try {
            // Check if file exists
            await fs.access(filePath);
        } catch (e) {
            console.log(`Skipping ${file}, not found`);
            continue;
        }

        let content = await fs.readFile(filePath, 'utf-8');
        const name = file.replace('.html', '');
        const folderName = name; // simple mapping

        // 1. Update Resources (CSS, JS, Images)
        content = content.replace(/href="styles\.css"/g, 'href="../styles.css"');
        content = content.replace(/src="script\.js"/g, 'src="../script.js"');
        content = content.replace(/src="images\//g, 'src="../images/');
        content = content.replace(/srcset="images\//g, 'srcset="../images/');
        content = content.replace(/content="images\//g, 'content="../images/');

        // 2. Update Header Logo Link
        content = content.replace(/href="index\.html"/g, 'href="../"');
        // Note: this might replace nav link too, which is fine: Home -> ../

        // 3. Update Nav Links and other internal links
        // We need to handle 'about.html', 'program.html' etc.
        // Regex to match href="filename.html"
        // And ensure we don't double replace if we run this script multiple times (though we shouldn't)

        filesToMigrate.forEach(f => {
            const fname = f.replace('.html', '');
            const regex = new RegExp(`href="${f}"`, 'g');
            content = content.replace(regex, `href="../${fname}/"`);

            // Also handle query params or anchors if simple regex missed them?
            // simpler regex above matches exact string.
            // Let's us a smarter regex for suffixes
            const regexSuffix = new RegExp(`href="${f}(#|\\?)`, 'g'); // e.g. about.html#team
            content = content.replace(regexSuffix, `href="../${fname}/$1`);
        });

        // 4. Set Active State in Navbar
        // Look for the link pointing to this page's new folder
        // e.g. href="../about/"
        // Add class="active". 
        // But beware of double class attribute or replacing existing class.
        // The nav links usually don't have a class except possibly 'active' which we want to set.
        // Current nav links: <li><a href="about.html">About Us</a></li>
        // Transformed above to: <li><a href="../about/">About Us</a></li>

        const myLinkRegex = new RegExp(`href="\.\.\/${folderName}\/"`);
        content = content.replace(myLinkRegex, `href="../${folderName}/" class="active"`);

        // 5. Create Directory and Move
        const folderPath = path.join(cwd, folderName);
        try {
            await fs.mkdir(folderPath, { recursive: true });
        } catch (e) { }

        const newFilePath = path.join(folderPath, 'index.html');
        await fs.writeFile(newFilePath, content, 'utf-8');

        console.log(`Migrated ${file} -> ${folderName}/index.html`);

        // 6. Delete old file
        await fs.unlink(filePath);
    }
}

migrate().catch(console.error);

File: migrate_urls.py

import os
import re

# Map of old filenames to new folder names
pages = {
    'about.html': 'about',
    'program.html': 'program',
    'team.html': 'team',
    'blog.html': 'blog',
    'contact.html': 'contact',
    'privacy.html': 'privacy',
    'blog-details.html': 'blog-details'
}

cwd = r'c:\Users\Thinkpad\Downloads\Heal_360'

def update_content(content, current_page_name, is_root=False):
    # Update CSS
    if is_root:
        # Root index.html doesn't need CSS/JS path changes, only links
        pass
    else:
        # Subpages need ../ prefix for resources
        content = content.replace('href="styles.css"', 'href="../styles.css"')
        content = content.replace('src="script.js"', 'src="../script.js"')
        content = content.replace('src="images/', 'src="../images/')
        content = content.replace('srcset="images/', 'srcset="../images/')
        content = content.replace('content="images/', 'content="../images/') # OG tags

    # Update Navigation Links
    # Map: 'index.html' -> '../' (if subpage) or './' (if root)
    # 'about.html' -> '../about/' (if subpage) or 'about/' (if root)
    
    # We will use regex to catch href="about.html" and href="about.html#..."
    
    def replace_link(match):
        url = match.group(1)
        if url.startswith('http') or url.startswith('#') or url.startswith('mailto:'):
            return f'href="{url}"'
        
        # Handle index.html
        if url == 'index.html' or url.startswith('index.html#'):
            suffix = url.replace('index.html', '')
            if is_root:
                return f'href="./{suffix}"' # ./ for root stability
            else:
                return f'href="../{suffix}"' # ../ for subpages to go up

        # Handle other pages
        for old_file, folder in pages.items():
            if url == old_file or url.startswith(old_file + '?') or url.startswith(old_file + '#'):
                suffix = url.replace(old_file, '')
                if is_root:
                     return f'href="{folder}/{suffix}"'
                else:
                     return f'href="../{folder}/{suffix}"'
        
        return f'href="{url}"' # No match found

    content = re.sub(r'href="([^"]+)"', replace_link, content)
    
    # Active State
    # Remove existing active classes
    # content = re.sub(r' class="active"', '', content) # Too dangerous, might remove from hamburger
    
    # Better: finding the specific link in nav-links and adding active class
    if not is_root and current_page_name:
        folder_name = pages[current_page_name]
        # Look for href="../folder_name/" and add class="active"
        # The link logic above transformed 'about.html' -> '../about/'
        
        target_href = f'../{folder_name}/'
        # Regex to find <li><a href="target_href"> and add class
        # Note: simplistic regex, assumes standard formatting we saw in view_file
        # Pattern: <li><a href="../about/">About Us</a></li>
        # We want: <li><a href="../about/" class="active">About Us</a></li>
        
        # Or if it already has a class? <a href="..." class="logo">
        
        pattern = f'href="{re.escape(target_href)}"'
        replacement = f'href="{target_href}" class="active"'
        content = re.sub(pattern, replacement, content)

    return content

# Process Subpages
for old_file, folder in pages.items():
    old_path = os.path.join(cwd, old_file)
    if not os.path.exists(old_path):
        print(f"Skipping {old_file}, not found")
        continue
        
    with open(old_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    new_content = update_content(content, old_file, is_root=False)
    
    folder_path = os.path.join(cwd, folder)
    os.makedirs(folder_path, exist_ok=True)
    
    new_file_path = os.path.join(folder_path, 'index.html')
    with open(new_file_path, 'w', encoding='utf-8') as f:
        f.write(new_content)
    
    print(f"Processed {old_file} -> {folder}/index.html")
    # Delete old file
    os.remove(old_path)

# Process Root index.html
index_path = os.path.join(cwd, 'index.html')
with open(index_path, 'r', encoding='utf-8') as f:
    content = f.read()

new_content = update_content(content, 'index.html', is_root=True)

with open(index_path, 'w', encoding='utf-8') as f:
    f.write(new_content)

print("Processed index.html")
